src/eureka/S3_data_reduction/optspex.py

```diff
@@ -511,40 +512,57 @@ def clean_median_flux(data, meta, log, median_thresh=5):
 
     # Create mask using all data quality flags
     mask = np.copy(data['mask'].values)
-    mask[np.where(data.dq > 0)] = 1
+    if meta.dqmask:
+        mask[np.where(data.dq % 2 == 1)] = 0
 
     # Compute median flux using masked arrays
     flux_ma = np.ma.masked_where(mask == 0, data.flux.values)
     medflux = np.ma.median(flux_ma, axis=0)
 
-    # Apply smoothing filter
-    ny, nx = medflux.shape
-    smoothflux = np.zeros((ny, nx))
-    for j in tqdm(range(ny)):
-        smoothflux[j] = smooth.medfilt(medflux.data[j], 31)
-
-    # Compute residuals
-    residuals = medflux - smoothflux
-
-    # Flag outliers
-    outliers = sigma_clip(residuals, sigma=median_thresh, maxiters=5, axis=1,
-                          cenfunc='median')
-
-    # Interpolate over bad pixels
-    clean_med = np.zeros((ny, nx))
-    xx = np.arange(nx)
-    for j in range(ny):
-        x1 = xx[~outliers.mask[j]]
-        goodmed = medflux[j][~outliers.mask[j]]
-        f = spi.interp1d(x1, goodmed, 'linear', fill_value='extrapolate')
-        # f = spi.UnivariateSpline(x1, goodmed, k=1, s=None)
-        clean_med[j] = f(xx)
-
-    data['medflux'] = (['y', 'x'], clean_med)
+    if meta.window_len > 1:
+        ny, nx = medflux.shape
+        # Interpolate over masked regions
+        interp_med = np.zeros((ny, nx))
+        xx = np.arange(nx)
+        for j in range(ny):
+            x1 = xx[~np.ma.getmaskarray(medflux[j])]
+            goodrow = medflux[j][~np.ma.getmaskarray(medflux[j])]
+            f = spi.interp1d(x1, goodrow, 'linear',
+                             fill_value='extrapolate')
+            # f = spi.UnivariateSpline(x1, goodmed, k=1, s=None)
+            interp_med[j] = f(xx)
```

Comment:
This needs to be done even if `meta.window_len == 1` (my MIRI case). If you found that there was no issue in doing this for your data, then we can just move it before the if statement. If it is detrimental for your data, then it could be moved to the else part of this if/else statement